app/src/integrationTest/kotlin/maru/app/MaruFollowerTest.kt

```diff
@@ -236,15 +237,140 @@ class MaruFollowerTest {
+          recipient = createAccount("another account"),
+          amount = Amount.ether(100),
+        )
+      }
+    }
+    checkNetworkStackBlocksProduced(validatorStack, 2 * blocksToProduce)
+
+    followerStack.setMaruApp(
+      maruFactory.buildTestMaruFollowerWithP2pPeering(
+        ethereumJsonRpcUrl = followerStack.besuNode.jsonRpcBaseUrl().get(),
+        engineApiRpc = followerStack.besuNode.engineRpcUrl().get(),
+        dataDir = followerStack.tmpDir,
+        validatorPortForStaticPeering = validatorStack.p2pPort,
+      ),
+    )
+    followerStack.maruApp.start()
+
+    checkValidatorAndFollowerBlocks(2 * blocksToProduce)
+  }
+
+  @Test
+  fun `Maru follower is able to complete syncing after disconnect peers`() {
+    val blocksToProduce = 20
+    repeat(blocksToProduce) {
+      transactionsHelper.run {
+        validatorStack.besuNode.sendTransactionAndAssertExecution(
+          logger = log,
+          recipient = createAccount("another account"),
+          amount = Amount.ether(100),
+        )
+      }
+    }
+
+    // This is here mainly to wait until block propagation is complete
+    checkValidatorAndFollowerBlocks(blocksToProduce)
+
+    val followerP2PNetwork = followerStack.maruApp.p2pNetwork()
+    val peers = followerP2PNetwork.getPeers()
+    peers.forEach {
+      followerP2PNetwork.dropPeer(it)
+    }
+
+    repeat(blocksToProduce) {
+      transactionsHelper.run {
+        validatorStack.besuNode.sendTransactionAndAssertExecution(
+          logger = log,
+          recipient = createAccount("another account"),
+          amount = Amount.ether(100),
+        )
+      }
+    }
+    checkNetworkStackBlocksProduced(validatorStack, 2 * blocksToProduce)
+    checkNetworkStackBlocksProduced(followerStack, blocksToProduce)
+    peers.forEach {
+      followerP2PNetwork.addPeer("${it.address}/p2p/${it.nodeId}")
+    }
+    checkNetworkStackBlocksProduced(followerStack, 2 * blocksToProduce)
+  }
+
   private fun checkValidatorAndFollowerBlocks(blocksToProduce: Int) {
     await
       .pollDelay(100.milliseconds.toJavaDuration())
-      // we need big timeout due to CI resources sometimes being slow
-      .timeout(1.minutes.toJavaDuration())
+      .timeout(30.seconds.toJavaDuration())
       .untilAsserted {
         val blocksProducedByQbftValidator = validatorStack.besuNode.getMinedBlocks(blocksToProduce)
         val blocksImportedByFollower = followerStack.besuNode.getMinedBlocks(blocksToProduce)
         assertThat(blocksImportedByFollower).isEqualTo(blocksProducedByQbftValidator)
       }
   }
+
+  private fun checkNetworkStackBlocksProduced(
+    stack: PeeringNodeNetworkStack,
+    blocksProduced: Int,
+  ) {
+    await
+      .pollDelay(100.milliseconds.toJavaDuration())
+      .timeout(30.seconds.toJavaDuration())
+      .untilAsserted {
+        val blocksOnStack = stack.besuNode.getMinedBlocks(blocksProduced)
+        assertThat(blocksOnStack.size).isEqualTo(blocksProduced)
+      }
+  }
```

Comment:
nit: just a small comment, I would recommend to extend this to 
```
private fun checkNetworkStackBlocksProduced(
    stacks: List<PeeringNodeNetworkStack>,
    blocksProduced: Int,
  ) 
```
so we could remove the use of `checkValidatorAndFollowerBlocks` and replace with `checkNetworkStackBlocksProduced`